gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java

```diff
@@ -219,18 +238,25 @@ private Identifier buildIdentifier(String rootId, String practiseCode) {
         return Optional.empty();
     }
 
-    private Extension buildProblemSignificance(CD linksetCode) {
-        Optional<String> code = Optional.ofNullable(linksetCode.getQualifier().get(0).getName().getCode());
-        Extension extension = new Extension();
-        extension.setUrl(PROBLEM_SIGNIFICANCE_URL);
 
-        if (code.isPresent() && code.get().equals(MAJOR_CODE)) {
-            extension.setValue(new CodeType("Major"));
-        } else {
-            extension.setValue(new CodeType("Minor"));
+    private Extension buildProblemSignificance(CD linkSetCode) {
+        Extension extension = new Extension()
+            .setUrl(PROBLEM_SIGNIFICANCE_URL);
+
+        return hasCode(linkSetCode) && MAJOR_CODE.equals(linkSetCode.getQualifier().get(0).getName().getCode())
+            ? extension.setValue(new CodeType(MAJOR_CODE_NAME)) : extension.setValue(new CodeType(MINOR_CODE_NAME));
+    }
+
+    private Optional<Annotation> generateAnnotationToMinor(CD linkSetCode) {
+        if (hasCode(linkSetCode) && !MAJOR_CODE.equals(linkSetCode.getQualifier().get(0).getName().getCode())) {
```

Comment:
Maybe it's worth creating a method for `hasCode(linkSetCode) && !MAJOR_CODE.equals(linkSetCode.getQualifier().get(0).getName().getCode()` and then using it in `buildProblemSignificance` method as well (just changing the order in the ternary operator)